src/file_resolver.rs

```diff
@@ -156,107 +181,57 @@ impl FileResolver {
     ) -> Result<ResolvedFile, String> {
         let path = PathBuf::from(filename);
 
-        let path = if let Ok(m) = path.strip_prefix("./") {
-            m.to_path_buf()
-        } else {
-            path
-        };
+        // relative path
```

Comment:
This comment confuses me: Aren't relative paths all paths except ones starting with a slash `/`? Where `./` and `../` are just special cases of relative paths.

Also, what about other OS, does `PathBuf::from(filename)` normalize the path?

Reply:
In Solidity, when you do `import "foo.sol";` then relative paths are not considered; that's one of things that this PR is fixing.

Reply:
> Also, what about other OS, does PathBuf::from(filename) normalize the path?

That just creates a `PathBuf` from a `String` or `OsString`.

Reply:
> In Solidity, when you do import "foo.sol"; then relative paths are not considered; that's one of things that this PR is fixing.

Sure, that's why I thought the comment could explain that (or maybe even better in the doc comment), because this isn't apparent by the code or by the comment. Sorry I wasn't entirely clear about this first hand.

> That just creates a PathBuf from a String or OsString.

I think that on Windows, backslashes are used for those special relative paths. It's not obvious to me what should happen in that case. or are we just saying that the import path must always be a unix path?

Reply:
Should all be fixed now